openverse_catalog/dags/providers/provider_api_scripts/brooklyn_museum.py

```diff
@@ -2,186 +2,155 @@
+            # https://www.brooklynmuseum.org/api/new-documentation/#definition-Image
+            size_type = image.get("largest_derivative", "")
+            for size in size_list:
+                if size.get("size", "") == size_type:
+                    height = size.get("height")
+                    width = size.get("width")
+        return height, width
+
+    @staticmethod
+    def _get_metadata(data):
+        return {
+            "accession_number": data.get("accession_number"),
+            "date": data.get("object_date"),
+            "description": data.get("description"),
+            "medium": data.get("medium"),
+            "credit_line": data.get("credit_line"),
+            "classification": data.get("classification"),
+        }
+
+    @staticmethod
+    def _get_creators(data):
+        artists_info = data.get("artists")
+        if isinstance(artists_info, list):
+            creators_list = (
+                artists.get("name")
+                for artists in artists_info
+                if artists.get("rank") == 1
             )
+            creator = next(creators_list, None)
+        else:
+            creator = None
+        return creator
+
+    @staticmethod
+    def _handle_object_data(data, license_url) -> list[dict]:
+        images = []
+        image_info = data.get("images")
+        if image_info is not None:
+            id_ = data.get("id", "")
+            title = data.get("title", "")
+            foreign_url = f"https://www.brooklynmuseum.org/opencollection/objects/{id_}"
+            metadata = BrooklynMuseumDataIngester._get_metadata(data)
+            creators = BrooklynMuseumDataIngester._get_creators(data)
+
+            for image in image_info:
+                foreign_id = image.get("id")
+                if foreign_id is None:
+                    continue
+                image_url = image.get("largest_derivative_url")
+                if image_url is None:
+                    continue
+                height, width = BrooklynMuseumDataIngester._get_image_sizes(image)
+                license_info = get_license_info(license_url=license_url)
+                images.append(
+                    {
+                        "foreign_landing_url": foreign_url,
+                        "image_url": image_url,
+                        "license_info": license_info,
+                        "foreign_identifier": foreign_id,
+                        "width": width,
+                        "height": height,
+                        "title": title,
+                        "meta_data": metadata,
+                        "creator": creators,
+                    }
+                )
+        return images
+
+    def get_record_data(self, data: dict) -> dict | list[dict] | None:
+        rights_info = data.get("rights_type")
+        license_url = self._get_license_url(rights_info)
+        if license_url is None:
+            return None
+        id_ = data.get("id")
```

Comment:
If we switch the `id` and `rights_info` checks here, we will return faster if `id_` is None.